Formula/m3u8_downloader.rb

```diff
@@ -0,0 +1,15 @@
+class M3u8Downloader < Formula
+  desc "Download m3u8 HLS streams, written in Rust"
+  homepage "https://github.com/DevNulPavel/m3u8_downloader"
+  url "https://github.com/DevNulPavel/m3u8_downloader/releases/download/v1.0.0/m3u8_downloader-universal.tar.gz"
+  sha256 "825ac395664ce89b48fd4d2672f551eec8eaa42e44962d979e2995a896695d4a"
+  license "MIT"
+
+  def install
+    bin.install "m3u8_downloader"
+  end
+
+  test do
+    assert_match "M3U8 downloader 1.0.0", shell_output("m3u8_downloader -V")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.